bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/internal/Ism8Handler.java

```diff
@@ -0,0 +1,183 @@
+                        try {
+                            if (dp.getId() == id) {
+                                dataPoint = dp;
+                                break;
+                            }
+                        } catch (Exception e) {
+                        }
+                    }
+
+                    if (dataPoint != null) {
+                        try {
+                            this.server.sendData(dataPoint.createWriteData(command));
+                        } catch (Exception e) {
+                            logger.error("Writting to ISM DataPoint '{}' failed. '{}'", dataPoint.getId(),
+                                    e.getMessage());
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void dispose() {
+        if (server != null) {
+            try {
+                logger.debug("Ism8: dispose");
+                server.stop();
+            } catch (Exception e) {
+            }
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    public void initialize() {
+        logger.debug("Ism8: Start initializing!");
+        config = getConfigAs(Ism8Configuration.class);
+        logger.debug("Ism8: PortNumber={}", config.getPortNumber());
+
+        try {
+            server = new Server(config.getPortNumber());
+            updateStatus(ThingStatus.UNKNOWN);
+
+            for (Channel channel : getThing().getChannels()) {
+                if (channel.getConfiguration().containsKey("id") && channel.getConfiguration().containsKey("type")) {
+                    int id = Integer.parseInt(channel.getConfiguration().get("id").toString());
+                    String type = channel.getConfiguration().get("type").toString();
+                    String description = channel.getLabel();
+                    server.addDataPoint(id, type, description);
+                } else {
+                    logger.error("Ism8: ID or type missing - Channel={}  Cfg={}", channel.getLabel(),
+                            channel.getConfiguration());
+                }
+
+                logger.debug("Ism8: Channel={}", channel.getConfiguration().toString());
+            }
+
+            updateStatus(ThingStatus.UNKNOWN);
+            server.addDataPointChangeListener(this);
+            scheduler.execute(() -> {
+                try {
+                    server.start();
+                } catch (Exception e) {
+
+                }
+            });
+
+            logger.debug("Finished initializing!");
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE);
+            logger.error("Ism8 Initialize: {}", e.getMessage());
+        }
+    }
+
+    @Override
+    public void channelLinked(ChannelUID channelUID) {
+        logger.debug("Ism8: Channel Link {}", channelUID.getId());
+    }
+
+    @SuppressWarnings("null")
```

Comment:
Please remove all warning suppressions from your code

Reply:
I removed the suppressions, but received afterwards a few warnings in Eclipse. There were null-checks in an IF statement, but I could only solve it by copying the object reference in a local variable. Please let me know if this is the right way (e.g. Ism8Handler.java line 61, 69, 78).